rewrite-core/src/main/java/org/openrewrite/TreeVisitor.java

```diff
@@ -54,7 +54,7 @@
  * @param <P> An input object that is passed to every visit method.
  */
 public abstract class TreeVisitor<T extends Tree, P> {
-    private Cursor cursor = new Cursor(null, Cursor.ROOT_VALUE);
+    Cursor cursor = new Cursor(null, Cursor.ROOT_VALUE);
```

Comment:
@jkschneider I suggest that we add the following few lines to the `Cursor(Cursor, Object)` constructor, so that we don't run into concurrency issues:
```java
        if (parent == null && value == ROOT_VALUE) {
            messages = new ConcurrentHashMap<>();
        }
```
What do you think?